stac_geoparquet/arrow/_api.py

```diff
@@ -81,39 +87,77 @@ def parse_stac_ndjson_to_arrow(
     Keyword Args:
         limit: The maximum number of JSON Items to use for schema inference
 
-    Yields:
-        Arrow RecordBatch with a single chunk of Item data.
+    Returns:
+        pyarrow RecordBatchReader with a stream of STAC Arrow RecordBatches.
     """
     # If the schema was not provided, then we need to load all data into memory at once
     # to perform schema resolution.
     if schema is None:
         inferred_schema = InferredSchema()
         inferred_schema.update_from_json(path, chunk_size=chunk_size, limit=limit)
         inferred_schema.manual_updates()
-        yield from parse_stac_ndjson_to_arrow(
+        return parse_stac_ndjson_to_arrow(
             path, chunk_size=chunk_size, schema=inferred_schema
         )
-        return
 
     if isinstance(schema, InferredSchema):
         schema = schema.inner
 
-    for batch in read_json_chunked(path, chunk_size=chunk_size):
-        yield stac_items_to_arrow(batch, schema=schema)
+    batches_iter = (
```

Comment:
This is probably fine, but I wanted to clarify a couple things:

1. Do we have 2 schemas here: in input schema (named `schema)` and an output schema (named `resolved_schema`)?
2. Could we somehow derive `resolved_schema` from *just* the input schema, and not from actual data? Something like `stac_items_to_arrow([], schema=schema)`? Or do we need actual items to figure out the resolved schema?

Reply:
1. Yes. It's the difference between the `StacJsonBatch` https://github.com/stac-utils/stac-geoparquet/blob/ab2701f7b47e66812066161cd1618f3d5276bdb4/stac_geoparquet/arrow/_batch.py#L36-L52 and `StacArrowBatch` https://github.com/stac-utils/stac-geoparquet/blob/ab2701f7b47e66812066161cd1618f3d5276bdb4/stac_geoparquet/arrow/_batch.py#L176-L180

	One is the schema of the _input data_, which is as close to the original STAC JSON as possible (only with geometry pre-coerced to WKB), and the other is the schema of the _output data_, after any STAC GeoParquet transformations.
2. No, not as it stands, and it's very annoying. I think the main blocker to this is that we transform the bounding box from an arrow List to an arrow Struct (which we do to take advantage of GeoParquet 1.1, which defines a bounding box struct column that can be used for predicate pushdown). However we don't know in advance whether the bounding box of each Item is 2D or 3D, and so we don't know in advance how many struct fields to create. 
	
	This also means that STAC conversion will fail on mixed 2D/3D input. Are there any real-world STAC collections that have mixed 2D/3D bounding boxes?

Reply:
OK thanks. We might consider revisiting this later but that all makes sense for now.

> Are there any real-world STAC collections that have mixed 2D/3D bounding boxes?

I think we only have one collection with 3D bounding boxes, and that should have 3D for each item.

Reply:
> However we don't know in advance whether the bounding box of each Item is 2D or 3D, and so we don't know in advance how many struct fields to create.

I suppose this is something we could keep track of in our `InferredSchema` class while we're doing a scan of the input data: keep track of whether bounding boxes are only 2D or only 3D or a mix of the two. 

Though if a user passed in their own schema (which describes the input, not the output data, and so describes `bbox` as a `List`), they'd also need to pass in whether the bbox is 2D or 3D